src/services/paras/ParasService.spec.ts

```diff
@@ -0,0 +1,241 @@
+	describe('ParasService.leaseInfo', () => {
+		it('Should return the correct leasing information', async () => {
+			const expectedResponse = {
+				at: expectedAt,
+				leases: [
+					{
+						account: '5CXFhuwT7A1ge4hCa23uCmZWQUebEZSrFdBEE24C41wmAF4N',
+						deposit: '1000000',
+					},
+					{
+						account: '5ESEa1HV8hyG6RTXgwWNUhu5fXvkHBfEJKjw3hKmde7fXdHQ',
+						deposit: '2000000',
+					},
+				],
+				paraLifeCycle: 'Onboarding',
+				onboardingAs: 'parachain',
+			};
+
+			const response = await parasService.leaseInfo(blockHash789629, paraId);
+
+			expect(sanitizeNumbers(response)).toMatchObject(expectedResponse);
+		});
+
+		it('Should return a null leases when length is equal to 0', async () => {
+			const emptyLeasesAt = () => Promise.resolve().then(() => []);
+
+			(mockApi.query.slots.leases.at as unknown) = emptyLeasesAt;
+
+			const expectedResponse = {
+				at: expectedAt,
+				leases: null,
+				paraLifeCycle: 'Onboarding',
+				onboardingAs: 'parachain',
+			};
+
+			const response = await parasService.leaseInfo(blockHash789629, paraId);
+
+			expect(sanitizeNumbers(response)).toMatchObject(expectedResponse);
+
+			(mockApi.query.slots.leases.at as unknown) = slotsLeasesAt;
+		});
+	});
+
+	describe('ParasService.leasesCurrent', () => {
+		it('Should return the correct entries for leasesCurrent', async () => {
+			const expectedResponse = {
+				at: expectedAt,
+				leasePeriodIndex: '39',
+				endOfLeasePeriod: '800000',
+				currentLeaseHolders: ['199', '200'],
+			};
+
+			const response = await parasService.leasesCurrent(blockHash789629, true);
+
+			expect(sanitizeNumbers(response)).toMatchObject(expectedResponse);
+		});
+
+		it('Should return the correct response exlcuding currentLeaseHolders', async () => {
+			const expectedResponse = {
+				at: expectedAt,
+				leasePeriodIndex: '39',
+				endOfLeasePeriod: '800000',
+				currentLeaseHolders: undefined,
+			};
+
+			const response = await parasService.leasesCurrent(blockHash789629, false);
+
+			expect(sanitizeNumbers(response)).toMatchObject(expectedResponse);
+		});
+	});
+
+	describe('ParasService.paras', () => {
+		it('Should return correct ParaLifecycles response', async () => {
+			const expectedResponse = {
+				at: expectedAt,
+				paras: [
+					{
+						paraId: '199',
+						paraLifeCycle: 'Onboarding',
+						onboardingAs: 'parachain',
+					},
+					{
+						paraId: '200',
+						paraLifeCycle: 'Parathread',
+						onboardingAs: undefined,
+					},
+				],
+			};
+
+			const response = await parasService.paras(blockHash789629);
+
+			expect(sanitizeNumbers(response)).toMatchObject(expectedResponse);
+		});
+	});
+
+	describe('ParasService.auctionsCurrent', () => {
+		it('Should return to correct data during an ongoing auction', async () => {
+			const expectedResponse = {
+				at: expectedAt,
+				beginEnd: '39',
+				finishEnd: '20039',
+				phase: 'ending',
+				auctionIndex: '4',
+				leasePeriods: ['1000', '1001', '1002', '1003'],
+				winning: [
```

Comment:
I am leaning towards us adjusting `winning` to return a vec with length equal to the slot ranges, so length 10. The rest could just be `Option::None`

Reply:
fixed